docs/API.md

```diff
@@ -0,0 +1,15 @@
+# API
+
+The Zonemaster GUI provides a small API to access specific resource and perform
+specific tasks.
+
+The supported paths are the following:
+
+* `domain_check`: the default path, access the input form
+* `domain_check/<domain>`: populate the input form with `<domain>` and launch
```

Comment:
Or shorter just `check`?

I can see that we want to add other parts to the `check`. We do not have to do it now, but I suggest that is we make room for it now. One possibility is to define that from first "/" to the end or next "/" there is one or more keys.

Keys are strings of 'a-z' and can take an value, if defined for the key. The value is a string of 'a-z0-9'. The value is separated from the key by '='. 

Keys are separated by ':'. 

Unknown keys can be ignored.

Now we only define 'check', 'result', 'test' and 'faq'. In the future we might add a key 'lang' that can be combined with 'check', e.g. `/check:lang=en/afnic.fr` or `/faq:lang=fr` or `/result:lang=fi/<test-id>`

Reply:
> Or shorter just check?

I kept the original path.

Why not keeping it simple and use query strings?
```
domain_check?lang=en
faq?lang=en
domain_check/afnic.fr?lang=fr&no-ipv4&ns=...&ds=...
```

> We do not have to do it now

I agree. This is left to a later work.

Reply:
> Why do you think "domain_check" is better than just "check"?

I didn't say it's better. I'm just saying that this is already the default path. So why not just reusing it for a start?

> That could be used for the API in general, couldn't it?

I think so yes.

> Is there any advantage of /domain_check/afnic.fr instead of /?domain_check=afnic.fr?

The only advantage is to build on what already exists.

Reply:
The goal here is not to define a whole API for the GUI. It is only to extend what we currently have to allow populating the input form from the URL and automatically launch the test. A side effect of this PR is to create a new API.md document.

Reply:
> I didn't say it's better. I'm just saying that this is already the default path. So why not just reusing it for a start?

I thought that this was about creating something new. How is "domain_check" the default? What would the consequence being of changing that to "check"?

> > Is there any advantage of /domain_check/afnic.fr instead of /?domain_check=afnic.fr?
> 
> The advantage I see is to build on what already exists.

Do you mean exist as "/result/" etc? But now we should have a broader perspective where we see that we in the future may want to add other elements, and then you suggest them to be added as "?KEY" or "?KEY=VALUE", which is reasonable. Why not base the new API on that? What would the consequence being of using that instead?

Reply:
> The goal here is not to define a whole API for the GUI. It is only to extend what we currently have to allow populating the input form from the URL and automatically launch the test. A side effect of this PR is to create a new API.md document.

I disagree that the API specification is a side effect. Instead of adding features ad-hoc we should specify first in a way that make the API stable for future extensions.

Reply:
> I thought that this was about creating something new.

The goal of this PR is to provide an easy way to launch a test from the URL.

> I disagree that the API specification is a side effect. Instead of adding features ad-hoc we should specify first in a way that make the API stable for future extensions.

This is not the goal of this PR.

Reply:
We could break out the creation of the API document to a separate PR and pause this PR until the API has been set.